### PR TITLE
More removal of version(unittest)

### PR DESCRIPTION
**Pull request by schveiguy**
See #6450

### CODE REVIEW

File: std/internal/cstring.d [hunk 1]
```diff
@@ -41,18 +41,6 @@ module std.internal.cstring;
 import std.range;
 import std.traits;
 
-version(unittest)
-@property inout(C)[] asArray(C)(inout C* cstr) pure nothrow @nogc @trusted
```
Comment (schveiguy) [inline std/internal/cstring.d]: Putting a trusted function like this in the global namespace is not a good idea.

File: std/socket.d [hunk 2]
```diff
@@ -122,14 +122,10 @@ else
 
 version(unittest)
 {
-    static assert(is(uint32_t == uint));
-    static assert(is(uint16_t == ushort));
```
Comment (schveiguy) [inline std/socket.d]: Note, `uint32_t` and `uint16_t` are nowhere else in this file. I have no idea what these were for so I removed them. If they are critical to have, they should go somewhere else.

File: std/socket.d [hunk 3]
```diff
@@ -454,6 +450,7 @@ class Protocol
 version(CRuntime_Bionic) {} else
 @safe unittest
 {
+    // import std.stdio : writefln;
```
Comment (schveiguy) [inline std/socket.d]: writefln is used in commented out lines below. I put in this comment because before it was covered by the version(unittest) block.

**Comment by schveiguy** [comment]
ping @wilzbach what is this circleci error in typecons? I can't figure out if it's something I did or not, but I didn't touch typecons.

**Comment by JackStouffer** [comment]
Random error, ignore it for the time being.

**Comment by schveiguy** [comment]
> Random error

Ugh. Now doc builder too.

**Comment by schveiguy** [comment]
Doc failure seems repeatable, but I don't know what it means. @wilzbach or @CyberShadow ?

**Comment by CyberShadow** [comment]
Er, I'm not sure @wilzbach worked on the spec testing stuff, maybe he knows?

Might be exposing a latent compiler bug through some edge case.

**Comment by wilzbach** [comment]
Looks really weird:

```
---
void main() {
import std.stdio;
        struct S { int x; float y; }

        int foo(S s) { return s.x; }

        foo( S(1, 2) ); // set field x to 1, field y to 2
        
}
---
/usr/bin/ld: error in __stdin.o(.eh_frame); no .eh_frame_hdr table will be created.
/usr/bin/ld: __stdin.o: invalid string offset 1096055152 >= 528 for section `.strtab'
/usr/bin/ld: unknown operator '(' in complex symbol
/usr/bin/ld: final link failed: Invalid operation
collect2: error: ld returned 1 exit status
Error: linker exited with status 1
```

I can try more testing tomorrow, but for now I have no idea why this is failing.

**Comment by schveiguy** [comment]
@wilzbach thanks for looking into it.

**Comment by wilzbach** [comment]
Regarding the DAutoTest failure -> https://github.com/dlang/dlang.org/pull/2345

**Pull request merged**